pkg/injector/patch_test.go

```diff
@@ -2,50 +2,133 @@ package injector
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"testing"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
 	mapset "github.com/deckarep/golang-set"
 	"github.com/golang/mock/gomock"
 	"github.com/google/uuid"
 	tassert "github.com/stretchr/testify/assert"
 	admissionv1 "k8s.io/api/admission/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
 
 	"github.com/openservicemesh/osm/pkg/certificate/providers/tresor"
 	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/constants"
 	k8s "github.com/openservicemesh/osm/pkg/kubernetes"
 	"github.com/openservicemesh/osm/pkg/tests"
 )
 
-var _ = Describe("Test all patch operations", func() {
+func TestCreatePatch(t *testing.T) {
+	assert := tassert.New(t)
 
-	// Setup all constants and variables needed for the tests
+	// Setup all variables and constants needed for the tests
 	proxyUUID := uuid.New()
 	const (
 		namespace = "-namespace-"
 		podName   = "-pod-name-"
 	)
 
-	Context("test createPatch() function", func() {
-		It("creates a patch", func() {
+	testCases := []struct {
+		name               string
+		namespace          *corev1.Namespace
+		rawExpectedPatches string
+	}{
+		{
+			name: "creates a patch",
+			namespace: &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			},
+			rawExpectedPatches: fmt.Sprintf(`[`+
+				// Add Envoy UID Label
```

Comment:
I feel using this extremely long string to verification is over-killing, and easy to break by other changes (e.g. what if we change the ENV feed to envoy container?).

I suggest, doing substring search in the `createPatch()` response (`rawPatch` in this case) should be sufficient.

Reply:
Looks good now.